cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -494,24 +509,36 @@ protected static JsonObject parseLoadStats(JsonArray brokerLoadBeforeArray, Json
 
         // There is no guarantee that the brokers are in the same order in both the before and after arrays.
         // Therefore we need to convert them into maps indexed by broker ID so we can align them later for the comparison.
-        Map<Integer, Map<String, Object>> loadBeforeMap = extractLoadParameters(brokerLoadBeforeArray);
+
+        Map<Integer, Map<String, Object>> loadBeforeMap = new HashMap<>();
+
+        if (brokerLoadBeforeArray != null) {
+            loadBeforeMap = extractLoadParameters(brokerLoadBeforeArray);
```

Comment:
We are taking care of that part in the next lines. In case the brokerLoadBeforeArray is empty we will get a `loadBeforeMap` which would be empty and can be handled in the next line ->
```
        if (!loadBeforeMap.isEmpty() && loadBeforeMap.size() != loadAfterMap.size()) {
            throw new IllegalArgumentException("Broker data was missing from the load before/after information");
        }
```

Reply:
But I mean, assume the above `if (brokerLoadBeforeArray != null)` condition is true, you enter the code block and `brokerLoadBeforeArray` is empty, what happens by executing `loadBeforeMap = extractLoadParameters(brokerLoadBeforeArray);` ??

Reply:
If the array is empty you will get an empty map `loadBeforeMap` from the `extractLoadParameters(brokerLoadBeforeArray`.

Reply:
I know :-D ... let me rephrase it. Having an empty array is going to be an issue? It will just return an empty map, and such empty map is well handled?

Reply:
Do we need to fix somehting here? As per me even if we get an empty map, that case is somehow handled in the later code. I can understand we don't have to parse any empty map so maybe I can also add an extra condition like `if (brokerLoadBeforeArray !=null && !brokerLoadBeforeArray.isEmpty()` then only we parse?